src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -452,3 +473,20 @@ void grpc_tls_certificate_provider_release(
   grpc_core::ExecCtx exec_ctx;
   if (provider != nullptr) provider->Unref();
 }
+
+grpc_status_code grpc_tls_certificate_key_match(
+    const char* private_key, const char* cert_chain, const char** error_details){
+  grpc_core::ExecCtx exec_ctx;
+  grpc_status_code code;
+  absl::Status match_status = grpc_core::PrivateKeyAndCertificateMatch(private_key, cert_chain).status();
+  if (!match_status.ok()) {
+    code = static_cast<grpc_status_code>(match_status.code());
+    *error_details = gpr_strdup(std::string(match_status.message()).c_str());
+  }else {
+    code = grpc_core::PrivateKeyAndCertificateMatch(private_key, cert_chain).value() ?
+             GRPC_STATUS_OK : static_cast<grpc_status_code>(absl::StatusCode::kInvalidArgument);
+    *error_details = grpc_core::PrivateKeyAndCertificateMatch(private_key, cert_chain).value() ?
```

Comment:
Under this implementation, we will need to invoke `grpc_core::PrivateKeyAndCertificateMatch` three times for the worst case. 
A question for you: how can we improve the code so that we only need one invocation for all cases?

Reply:
I changed the code so as to call it just once